corehq/apps/locations/static/locations/js/location_drilldown.js

```diff
@@ -63,7 +63,13 @@ hqDefine('locations/js/location_drilldown', [
             if (model.selected_path().length !== model.location_types.length &&
                 model.selected_path.indexOf(loc) === -1 &&
                 model.selected_path().length < model.max_drill_depth) {
-                model.selected_path.push(loc);
+
+                var levelAlreadyInPath = false;
+                model.selected_path().forEach(function(selected) {
+                    if (!levelAlreadyInPath) levelAlreadyInPath = loc.depth === selected.depth;
+                });
```

Comment:
I think this would be more clear using `some`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

`levelAlreadyInPath = model.selected_path().some(function(selected) { return loc.depth === selected.depth; })`